File: backend/app/api/v1/play_records.py
"""播放记录 API"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import get_db
from app.schemas import play_records as schemas_play_records
from app.schemas.response import ApiResponse, PaginatedResponse, error, success
from app.services import play_records as services_play_records
from app.utils.exception import ServiceException


router = APIRouter(prefix="/play-records", tags=["play-records"])


@router.post(
    "/list",
    response_model=ApiResponse[
        PaginatedResponse[schemas_play_records.PlayRecordItemResponse]
    ],
)
def query_play_record_list(
    data: schemas_play_records.PlayRecordListRequest,
    db: Session = Depends(get_db),
):
    """分页查询播放记录列表"""
    try:
        result = services_play_records.get_play_record_list(data.page, data.page_size, db)
        return success(data=result)
    except ServiceException as exc:
        return error(code=exc.code, message=exc.message)


@router.post(
    "/detail",
    response_model=ApiResponse[schemas_play_records.PlayRecordItemResponse | None],
)
def query_play_record_detail(
    data: schemas_play_records.PlayRecordDetailRequest,
    db: Session = Depends(get_db),
):
    """查询单个视频的播放记录"""
    try:
        result = services_play_records.get_play_record_detail(
            data.site_id,
            data.vod_id,
            db,
        )
        return success(data=result)
    except ServiceException as exc:
        return error(code=exc.code, message=exc.message)


@router.post(
    "/upsert",
    response_model=ApiResponse[schemas_play_records.PlayRecordItemResponse],
)
def upsert_play_record(
    data: schemas_play_records.PlayRecordUpsertRequest,
    db: Session = Depends(get_db),
):
    """写入播放记录"""
    try:
        result = services_play_records.upsert_play_record(data, db)
        return success(data=result)
    except ServiceException as exc:
        return error(code=exc.code, message=exc.message)


@router.post(
    "/delete",
    response_model=ApiResponse[schemas_play_records.PlayRecordDeleteResponse],
)
def delete_play_record(
    data: schemas_play_records.PlayRecordDeleteRequest,
    db: Session = Depends(get_db),
):
    """删除单条播放记录"""
    try:
        result = services_play_records.delete_play_record(data.record_id, db)
        return success(data=result)
    except ServiceException as exc:
        return error(code=exc.code, message=exc.message)


@router.post(
    "/clear",
    response_model=ApiResponse[schemas_play_records.PlayRecordDeleteResponse],
)
def clear_play_records(
    _request: schemas_play_records.PlayRecordClearRequest,
    db: Session = Depends(get_db),
):
    """清空播放记录"""
    try:
        result = services_play_records.clear_play_records(db)
        return success(data=result)
    except ServiceException as exc:
        return error(code=exc.code, message=exc.message)

File: frontend/src/api/playRecords.js
import request from '@/utils/request'

/**
 * 分页查询播放记录列表
 * @param {Object} params - 分页参数
 * @returns {Promise} 播放记录列表
 */
export function getPlayRecords({ page = 1, pageSize = 20 } = {}) {
  return request.get('/api/v1/play-records', {
    params: { page, page_size: pageSize }
  })
}

/**
 * 查询单个视频的播放记录
 * @param {Object} params - 站点与视频标识
 * @returns {Promise} 播放记录，不存在时 data 为 null
 */
export function getPlayRecordDetail({ siteId, vodId }) {
  return request.get('/api/v1/play-records/detail', {
    params: { site_id: siteId, vod_id: vodId }
  })
}

/**
 * 写入播放记录（同一视频自动更新）
 * @param {Object} record - 播放记录内容
 * @returns {Promise} 写入结果
 */
export function upsertPlayRecord({
  siteId,
  vodId,
  title,
  thumbnail,
  keyword,
  lineName,
  episodeIndex,
  episodeName,
  positionSeconds,
  durationSeconds
}) {
  return request.put('/api/v1/play-records', {
    site_id: siteId,
    vod_id: vodId,
    title,
    thumbnail: thumbnail || '',
    keyword,
    line_name: lineName || '',
    episode_index: episodeIndex || 0,
    episode_name: episodeName || '',
    position_seconds: Math.floor(positionSeconds || 0),
    duration_seconds: Math.floor(durationSeconds || 0)
  })
}

/**
 * 删除单条播放记录
 * @param {number} recordId - 播放记录 ID
 * @returns {Promise} 删除结果
 */
export function deletePlayRecord(recordId) {
  return request.delete(`/api/v1/play-records/${recordId}`)
}

/**
 * 清空播放记录
 * @returns {Promise} 清空结果
 */
export function clearPlayRecords() {
  return request.delete('/api/v1/play-records')
}
